include/oneapi/dpl/pstl/zip_view_impl.h

```diff
@@ -0,0 +1,505 @@
+        {
+            return it += n;
+        }
+
+        friend constexpr iterator
+        operator+(difference_type n, iterator it) requires all_random_access<Const, Views...>
+        {
+            return it += n;
+        }
+
+        friend constexpr iterator
+        operator-(iterator it, difference_type n) requires all_random_access<Const, Views...>
+        {
+            return it -= n;
+        }
+
+        friend constexpr decltype(auto) iter_move(const iterator& x) noexcept(
+            (noexcept(std::ranges::iter_move(std::declval<const std::ranges::iterator_t<__maybe_const<Const, Views>>&>())) && ...) &&
+            (std::is_nothrow_move_constructible_v<std::ranges::range_rvalue_reference_t<__maybe_const<Const, Views>>> && ...))
+        {
+            auto __tr = [](auto&&... __args) -> decltype(auto) {
+                return rvalue_reference_type(std::forward<decltype(__args)>(__args)...);
+            };
+            return apply_to_tuple(__tr, std::ranges::iter_move, x.current_);
+
+        }
+
+        friend constexpr void iter_swap(const iterator& x, const iterator& y) noexcept(
+            (noexcept(std::ranges::iter_swap(std::declval<const std::ranges::iterator_t<__maybe_const<Const, Views>>&>(),
+                                        std::declval<const std::ranges::iterator_t<__maybe_const<Const, Views>>&>())) && ...))
+          requires(std::indirectly_swappable<std::ranges::iterator_t<__maybe_const<Const, Views>>> && ...)
+        {
+            bi_apply_to_tuple(std::ranges::iter_swap, x.current_, y.current_);
+        }
+
+      private:
+        template <std::size_t... In>
+        constexpr bool
+        compare_equal(iterator y, std::index_sequence<In...>) const
+        {
+            return ((std::get<In>(current_) == std::get<In>(y.current_)) || ...);
+        }
+
+        template <typename SentinelsTuple, std::size_t... In>
+        constexpr bool
+        compare_with_sentinels(const SentinelsTuple& sentinels, std::index_sequence<In...>) const
+        {
+            return ((std::get<In>(current_) == std::get<In>(sentinels)) || ...);
+        }
+        
+        template <typename SentinelsTuple, std::size_t... In>
+        constexpr std::common_type_t<std::ranges::range_difference_t<__maybe_const<Const, Views>>...>        
+        distance_to_sentinels(const SentinelsTuple& sentinels, std::index_sequence<In...>) const
+        {
+            return std::ranges::min({difference_type(std::get<In>(current_) - std::get<In>(sentinels))...}, std::less{},
+                                    [](auto a){ return std::abs(a);});
+        }
+        
+        template <std::size_t... In>
+        constexpr std::common_type_t<std::ranges::range_difference_t<__maybe_const<Const, Views>>...>
+        distance_to_it(const iterator it, std::index_sequence<In...>) const
+        {
+            return std::ranges::min({difference_type(std::get<In>(it.current_) - std::get<In>(current_))...}, std::less{},
```

Comment:
It seems like the function `x.distance_to(y)` calculates a distance `(y - x)` and `x.distance_to_sentinels(y)` calculates `(x-y)`. I think we need to define the what is distance and be consistent.

Reply:
The arithmetic works correct because we use `distance_to_it` and `distance_to_sentinels` in an opposite way:

`x.operator-(iterator y)` calls `y.distance_to_it(x)` which calculates minimal of all `xi - yi`.
`x.operator-(sentinel y)` calls `x.distance_to_sentinels(y)` which also calculates minimal `xi - yi`.

As a result, we have the same calculations for both functions but they are not consistent with each other. I propose to use `x.distance_to(y)` or `y.distance_to(x)` consistent for both  functions.

Reply:
To eliminate any misunderstanding and confusing what is "distance" between to iterators or between iterator and sentinel, I would propose to remove `distance_to_it` and `distance_to_sentinels` private methods and write the impl code directly within `operator-` for iterator and `sentinel`.

Reply:
the code was updated.